pygbe/class_initialization.py

```diff
@@ -56,14 +56,13 @@ def initialize_field(filename, param, field=None):
     """
 
     if not field:
-        field = readFields(filename)
-
-    field['LorY'] = [int(i) if i != 'NA' else 0 for i in field['LorY']]
-    field['E'] = [complex(i) if 'j' in i else param.REAL(i) if i != 'NA' else 0
-         for i in field['E']]
-    field['kappa'] = [param.REAL(i) if i != 'NA' else 0 for i in field['kappa']]
-    field['pot'] = [int(i) for i in field['pot']]
-    field['coulomb'] = [int(i) for i in field['coulomb']]
+        field = read_fields(filename)
+
+    for key in ['E', 'kappa']:
+        for i, e in enumerate(field[key]):
+            if not isinstance(e, complex):
```

Comment:
Won't this line turn the value into `param.REAL` even in the case we have `NA` ? 
Also `kappa` is never `complex` why are we putting it here? I mean we will always been checking for a condition that never applies.